subprojects/solver/solver-z3/src/main/java/hu/bme/mit/theta/solver/z3/Z3TermTransformer.java

```diff
@@ -175,6 +154,14 @@ private <I extends Type, E extends Type> Expr<?> createIndexValueArrayLitExpr(I
 
     ////////
 
+    private Expr<?> transform(final com.microsoft.z3.Expr term, final Model model,
+                              final List<Decl<?>> vars, final Type type) {
+        if (type instanceof EnumType enumType) {
+            return transformEnumLit(term, enumType);
+        }
```

Comment:
Can we refactor it to resemble how we handle arrays or function literals? `toFuncLitExpr`, `toArrayLitExpr`

Reply:
These lines are removed in 0aa36d3